reV/supply_curve/point_summary.py

```diff
@@ -475,17 +476,58 @@ def _agg_data_layer_method(data, excl_mult, method):
             elif method.lower() == 'sum':
                 data = data.sum()
             elif method.lower() == 'category':
-                data = {category: excl_mult[(data == category)].sum()
-                        for category in np.unique(data)}
+                data = SupplyCurvePointSummary._agg_data_layer_category(
+                    data, excl_mult)
             else:
                 e = ('Cannot recognize data layer agg method: '
-                     '"{}". Can only do mean, mode, sum, or category_count.'
+                     '"{}". Can only do mean, mode, sum, or category.'
                      .format(method))
                 logger.error(e)
                 raise ValueError(e)
 
         return data
 
+    def _agg_data_layer_category(data, excl_mult):
```

Comment:
What about having this class return the dictionary, and then add a staticmethod to jump to str that you'd run when you're adding the data to the DataFrame. Long story short, I think it would be nice to be able to get the original dict